bin/main.ml

```diff
@@ -61,17 +61,33 @@ module Prompt = struct
     | true -> Lwt_io.printl "I: reading from stdin. Please type the secret and then do Ctrl+d twice to terminate input"
     | false -> Lwt.return_unit
 
-  let read_input_from_stdin () = Lwt_io.(read stdin)
-
-  let rec input_and_validate_loop ?(transform = fun x -> x) get_secret_input =
-    let%lwt input = get_secret_input () in
-    let secret = transform input in
+  let read_input_from_stdin ?initial:_ () = Lwt_io.(read stdin)
+
+  let rec input_and_validate_loop ?(transform = fun x -> x) ?initial get_secret_input =
+    let remove_trailing_newlines s =
+      let rec loop s =
+        match String.ends_with ~suffix:"\n" s with
+        | true -> loop (String.sub s 0 (String.length s - 1))
+        | false -> s
+      in
+      loop s
+    in
+    let%lwt input = get_secret_input ?initial () in
+    let input = transform input in
+    (* Remove bash commented lines from the secret and any trailing newlines *)
```

Comment:
Should we also remove all suffix comments? What about lines that start with `#` which are not a comment?  

I'm guessing the use-case for this is to remove the format explainer. In that case we can leave as is. Just checking.